### PR TITLE
Upgrade old packages scanned from snyk

### PR DESCRIPTION
**Pull request by rohit-dua**
### Description

Old packages with possible vulnerabilities are upgraded. (as tested on snyk.io)

### Related issues

- Fixes https://github.com/celo-org/celo-labs/issues/3

### Tested

 Ran local unit tests.

### Backwards compatibility

`yarn resolutions` is used to upgrade the versions, since some vulnerable versions(with new bugs) are still being used in public libraries.

### CODE REVIEW

**Review by jmrossy — CHANGES_REQUESTED** [review]
Some feedback here:
1. Please fill out the Test and Backwards Compat sections for PR descriptions.
2. Use the assignees field to indicate who should next take action on it (we do this as we bounce PRs back and forth between author and reviewer).
3. include other devs as necessary (this affects the faucet and cli too so @mcortesi, @ashishb and @trianglesphere should be included)
4. We try to keep all versions of dependencies consistent across our diff sub-packages. So if you're changing it in one package.json, try to match those changes elsewhere.

File: packages/faucet/package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "eth-lib": "^0.2.8",
     "firebase": "^6.2.2",
     "firebase-admin": "^7.0.0",
-    "firebase-functions": "^2.2.2",
+    "firebase-functions": "^2.3.1",
```
Comment (trianglesphere) [inline packages/faucet/package.json:27]: Does this match what is in `packages/verification-pool-api`? We try to keep versions the same across different packages.

Comment (rohit-dua) [reply packages/faucet/package.json:27]: Sure. Updated both packages to use version 2.3.1

File: packages/cli/package.json [hunk 2]
```diff
@@ -41,26 +41,27 @@
...
     "cli-table": "^0.3.1",
-    "cli-ux": "^5.3.1",
+    "cli-ux": "5.3.1",
     "debug": "^4.1.1",
     "elliptic": "^6.4.1",
     "events": "^3.0.0",
     "firebase": "^6.2.4",
     "fs-extra": "^8.1.0",
     "path": "^0.12.7",
+    "tar": "4.4.10",
```
Comment (jmrossy) [inline packages/cli/package.json]: Why add a dependency here?

File: packages/cli/package.json [hunk 3]
```diff
@@ -41,26 +41,27 @@
     "bn.js": "^5.0.0",
     "chalk": "^2.4.2",
     "cli-table": "^0.3.1",
-    "cli-ux": "^5.3.1",
+    "cli-ux": "5.3.1",
```
Comment (jmrossy) [inline packages/cli/package.json]: Why have you locked the version to 5.3.1?

File: packages/verification-pool-api/package.json [hunk 4]
```diff
@@ -28,7 +28,7 @@
     "ethereumjs-util": "^5.2.0",
     "express": "^4.16.4",
     "firebase-admin": "^7.0.0",
-    "firebase-functions": "^2.2.0",
+    "firebase-functions": "^2.3.1",
```
Comment (jmrossy) [inline packages/verification-pool-api/package.json:30]: Please redeploy the pool to ensure it still works after this update. Deploying to Int should be fine as we're not using it atm

File: packages/protocol/package.json [hunk 5]
```diff
@@ -58,9 +58,9 @@
     "solidity-bytes-utils": "^0.0.7",
     "solidity-coverage": "rotcivegaf/solidity-coverage#5875f5b7",
     "truffle": "5.0.19",
-    "truffle-artifactor": "^3.0.7",
+    "truffle-artifactor": "^4.0.4",
```
Comment (jmrossy) [inline packages/protocol/package.json]: @mcortesi @asaj FYI

File: packages/verification-pool-api/package.json (hunk 4 above)
Comment (rohit-dua) [reply packages/verification-pool-api/package.json:30]: test deployed it. It works fine.

File: packages/cli/package.json (hunk 3 above)
Comment (rohit-dua) [reply packages/cli/package.json]: I used snyk tool to fix the versions. I guess it locked the version. I've changed that.

**Review by jmrossy — COMMENTED** [review]
Still waiting on details about how these were tested

**Comment by rohit-dua** [comment]
The packages were tested locally and after deploying:
deploy: `verification-pool-api` and `faucet` and `blockchain-api`
celotool: local testing
mobile: local app testing

File: packages/cli/package.json [hunk 6]
```diff
@@ -53,11 +53,12 @@
...
     "@types/bip32": "^1.0.1",
     "@types/bip39": "^2.4.2",
-    "@types/elliptic": "^6.4.9",
-    "@types/mocha": "^5.2.7",
     "@types/cli-table": "^0.3.0",
     "@types/debug": "^4.1.4",
+    "@types/elliptic": "^6.4.9",
     "@types/fs-extra": "^8.0.0",
+    "@types/jest": "^24.0.15",
+    "@types/mocha": "^5.2.7",
```
Comment (mcortesi) [inline packages/cli/package.json]: we don't use mocha in cli

File: packages/cli/package.json [hunk 7]
```diff
@@ -53,11 +53,12 @@
     "@celo/dev-cli": "^2.0.3",
     "@types/bip32": "^1.0.1",
     "@types/bip39": "^2.4.2",
-    "@types/elliptic": "^6.4.9",
-    "@types/mocha": "^5.2.7",
     "@types/cli-table": "^0.3.0",
     "@types/debug": "^4.1.4",
+    "@types/elliptic": "^6.4.9",
     "@types/fs-extra": "^8.0.0",
+    "@types/jest": "^24.0.15",
```
Comment (mcortesi) [inline packages/cli/package.json]: this already lives in root's package.json

**Review by mcortesi — DISMISSED** [review]

**Comment by rohit-dua** [comment]
The faucet is tested and working.

**Comment by rohit-dua** [comment]
Tested with integration network:
deploy: verification-pool-api and faucet and blockchain-api
celotool: local testing
mobile: local app testing with faucet from local web

**Review by jmrossy — APPROVED** [review]
Ship it!

**Pull request merged**